### PR TITLE
[CIRCLE-38658] Add DD rum integration

### PR DESCRIPTION
**Pull request by rsylvian**
Ticket: [CIRCLE-38658](https://circleci.atlassian.net/browse/CIRCLE-38658)

# Description
- Add `services/rum.js` to integrate with DD Rum
- Add tests for both `rum.js` and `utils.js`

# Reasons
We want to have DD RUM running for Doc so we can measure performances for users and understand better what they experience.

### CODE REVIEW

File: src-js/services/rum.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { datadogRum } from '@datadog/browser-rum';
+import { isProduction, isUnsupportedBrowser } from '../utils';
+
+export const getDefaultOptions = () => ({
+  applicationId: '2a3e19ca-297e-4a53-a2cf-082e827d13d3',
+  clientToken: 'pubdffb4b4dd07f1f43b42276901d58da91',
```
Comment (rsylvian) [inline src-js/services/rum.js:6]: These are public keys FYI

File: src-js/utils.js [hunk 2]
```diff
@@ -1,3 +1,10 @@
 export function isProduction() {
   return window.location.origin === 'https://circleci.com';
 }
+
+export function isUnsupportedBrowser() {
+  if (typeof window === 'undefined') return true;
+  const ua = window?.navigator?.userAgent;
+  const msie = ua.indexOf('MSIE ');
+  return msie > 0;
```
Comment (tshen815) [inline src-js/utils.js:9]: what foes `msie` stand for in this instance?

File: src-js/services/rum.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  site: 'datadoghq.com',
+  service: 'circleci-docs',
+  env: isProduction() ? 'production' : 'development',
+  trackInteractions: false,
```
Comment (tshen815) [inline src-js/services/rum.js:10]: Im curious, wouldn't we want `trackInteractions` interactions to be `true` to help monitor user clicks in our app in combination with setting a [custom action name](https://docs.datadoghq.com/real_user_monitoring/browser/tracking_user_actions/?tab=npm#declaring-a-name-for-click-actions) or implementing [global scrubbing](https://docs.datadoghq.com/real_user_monitoring/browser/modifying_data_and_context/?tab=npm) to not release sensitive data?

File: src-js/utils.js (hunk 2 above)
Comment (rsylvian) [reply src-js/utils.js:9]: **M**icro**S**oft **I**nternet **E**xplorer 😃

File: src-js/services/rum.js (hunk 3 above)
Comment (rsylvian) [reply src-js/services/rum.js:10]: Probably not. These settings are directly coming from the one chosen on a company level in: https://github.com/circleci/frontplane/blob/80b07b1f3b7db7b6d541c2a1c59f617d02a96369/packages/real-user-monitoring/src/utils.ts#L25-L26

File: src-js/utils.test.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+      delete window.location;
+      window.location = new URL('https://circleci.com');
+
+      expect(utils.isProduction()).toBe(true);
+
+      window.location = originalLocation;
+    });
+  });
+
+  describe('isUnsupportedBrowser', () => {
```
Comment (ccgeller) [inline src-js/utils.test.js:25]: well done!

**Review by ccgeller — APPROVED** [review]
Bit unsure how to test with datadog but code seems to look fine and tests pass! 🥳🤩

File: src-js/services/rum.js (hunk 3 above)
Comment (tshen815) [reply src-js/services/rum.js:10]: thats a good point, makes sense!

**Review by tshen815 — APPROVED** [review]
looks good!

**Review by nramabad — APPROVED** [review]

**Pull request merged**